client/src/components/RegisterModal.vue

```diff
@@ -0,0 +1,266 @@
+<script setup lang="ts">
+import { ref, defineEmits } from 'vue'
+import { useDisplay } from 'vuetify'
+
+const { xs } = useDisplay()
+
+const formData = ref({
+  username: '',
+  email: '',
+  firstName: '',
+  lastName: '',
+  dateOfBirth: '',
+  genderId: null,
+  indigenousTIS: null,
+  password: '',
+  confirmPassword: '',
+})
+
+const dialog = ref(false)
+const emit = defineEmits(['close'])
+
+const openDialog = () => {
+  dialog.value = true
+}
+
+const closeDialog = () => {
+  dialog.value = false
+  emit('close')
+}
+
+const submitForm = () => {
+  closeDialog()
+}
+</script>
+
+<template>
+  <div>
+    <v-btn
```

Comment:
Can you move this button out of the component? For demonstration purposes, you can put it in ``HomeView.vue`` instead.